Src/FluentAssertions/Specialized/AssemblyAssertions.cs

```diff
@@ -139,6 +141,75 @@ public AndConstraint<AssemblyAssertions> Reference(Assembly assembly, string bec
         return new AndWhichConstraint<AssemblyAssertions, Type>(this, foundType);
     }
 
+    /// <summary>Asserts that the assembly is unsigned.</summary>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+    /// </param>
+    public AndConstraint<AssemblyAssertions> BeUnsigned(string because = "", params object[] becauseArgs)
+    {
+        bool success = Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject is not null)
+            .FailWith("Expected assembly to be unsigned{reason}, but {context:assembly} is <null>.");
+
+        if (success)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject.GetName().GetPublicKey() is not { Length: > 0 })
+                .FailWith(
+                    "Did not expect the assembly {0} to be signed{reason}, but it is.", Subject.FullName);
+        }
+
+        return new(this);
+    }
+
+    /// <summary>Asserts that the assembly is signed with the specified public key.</summary>
+    /// <param name="publicKey">
+    /// The base-16 string representation of the public key.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="publicKey"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="publicKey"/> is empty.</exception>
+    public AndConstraint<AssemblyAssertions> BeSignedWithPublicKey(string publicKey, string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNullOrEmpty(publicKey);
+
+        bool success = Execute.Assertion
+          .BecauseOf(because, becauseArgs)
+          .ForCondition(Subject is not null)
+          .FailWith("Expected assembly to have public key {0}{reason}, but {context:assembly} is <null>.", publicKey);
+
+        if (success)
+        {
+            var bytes = Subject.GetName().GetPublicKey();
```

Comment:
🤔 According to the MSDN docs, `GetPublicKey` can return `null` (and my IDE agrees).